x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -381,6 +384,18 @@ private static Failure validateUnsignedLongNegation(Neg neg) {
         return null;
     }
 
+    /**
+     * Makes sure that spatial types do not appear in sorting contexts.
+     */
+    private static void checkForSortOnSpatialTypes(LogicalPlan p, Set<Failure> localFailures) {
+        p.forEachDown(OrderBy.class, o -> o.order().forEach(agg -> agg.forEachUp(Attribute.class, fa -> {
```

Comment:
The outer loop already does forDown, so no need for you to do it. Also Order is an expression so you don't have to unroll the list; instead do:
```
if (p instanceof OrderBy ob) {
    ob.forEachExpressionOnly(Attribute, fa -> ...)
}
```

Similar to what the other methods are doing.

Reply:
I had copied the similar code from the SQL module, so did not think about the duplicate forEachDown, but I've removed it now. The code has more lines, but is more readable and no longer has the duplicated forEachDown.